net_io.c

```diff
@@ -1109,6 +1115,28 @@ static void moveNetClient(struct client *c, struct net_service *new_service)
     c->service = new_service;
 }
 
+static int handleFaupCommand(struct client *c, char *p) {
+    if (p == NULL)
+        return 0;
+
+    MODES_NOTUSED(c);
+    char* msg_field;
+    msg_field = strtok (p, "\t");
+
+    // Traverse through message for commands
+    while (msg_field != NULL) {
+        if (strcmp(msg_field, "upload_rate_multiplier") == 0) {
+            msg_field = strtok (NULL, "\t");
+            fprintf(stderr, "Adjusting message rate to FlightAware by %sx\n", msg_field);
+            Modes.faup_rate_multiplier = atof(msg_field);
```

Comment:
Test the return value of `atof` and only assign to `faup_rate_multiplier` if it's sensible (e.g. 0 < rate < 100)